projects/schematics/src/shared/utils/new-module-utils.ts

```diff
@@ -191,12 +189,36 @@ function addToModuleInternal(
   return createdNode;
 }
 
-function elementExists(
+function isDuplication(
   initializer: ArrayLiteralExpression,
-  moduleToCheck: string
+  propertyName: 'imports' | 'exports' | 'declarations' | 'providers',
+  content: string
 ): boolean {
+  if (propertyName === 'providers') {
+    const normalizedContent = normalizeConfiguration(content);
+    const configs = getSpartacusProviders(initializer.getSourceFile());
+    for (const config of configs) {
+      const normalizedConfig = normalizeConfiguration(config);
+      if (normalizedContent === normalizedConfig) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  return isTypeTokenDuplicate(initializer, content);
```

Comment:
Wasn't the case for `imports`, `exports`, `declarations` properly handled by the ts-morph before?

Reply:
Some cases where handled, and some not. Now we are sure that everything is covered when we normalize tokens.